installer/tue-install-impl.bash

```diff
@@ -556,28 +648,127 @@ function tue-install-ppa
 
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
-function tue-install-pip
+function tue-install-ppa-now
 {
-    tue-install-debug "tue-install-pip $*"
+    tue-install-debug "tue-install-ppa-now $*"
+
+    local PPA_ADDED=""
+    # shellcheck disable=SC2048
+    for ppa in $*
+    do
+        if ! grep -h "^deb.*${ppa#ppa:}" /etc/apt/sources.list.d/* 1>/dev/null 2>&1
+        then
+            tue-install-system-now software-properties-common
+            tue-install-info "Adding ppa: $ppa"
+            sudo add-apt-repository --yes "$ppa"
+            PPA_ADDED=true
+        else
+            tue-install-debug "$ppa is already added previously"
+        fi
+    done
+    if [ -n "$PPA_ADDED" ]
+    then
+        tue-install-debug "Updating apt-get"
+        sudo apt-get update -qq
+    fi
+}
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+function _tue-install-pip
+{
+    local pv=$1
+    shift
+    tue-install-debug "tue-install-pip${pv} $*"
 
     if [ -z "$1" ]
     then
-        tue-install-error "Invalid tue-install-pip call: needs package as argument."
+        tue-install-error "Invalid tue-install-pip${pv} call: needs package as argument."
     fi
-    tue-install-debug "Adding $1 to pip list"
-    TUE_INSTALL_PIPS="$1 $TUE_INSTALL_PIPS"
+    tue-install-debug "Adding $1 to pip${pv} list"
+    local list=TUE_INSTALL_PIP"${pv}"S
+    # shellcheck disable=SC2140
+    declare -g "$list"="$1 ${!list}"
+}
+
+# Needed for backward compatibility
+function tue-install-pip
+{
+    _tue-install-pip "2" "$@"
+}
+
+function tue-install-pip2
+{
+    _tue-install-pip "2" "$@"
 }
 
 function tue-install-pip3
 {
-    tue-install-debug "tue-install-pip3 $*"
+    _tue-install-pip "3" "$@"
+}
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+function _tue-install-pip-now
+{
+    local pv=$1
+    shift
+    tue-install-debug "tue-install-pip${pv}-now $*"
 
     if [ -z "$1" ]
     then
-        tue-install-error "Invalid tue-install-pip3 call: needs package as argument."
+        tue-install-error "Invalid tue-install-pip${pv}-now call: needs package as argument."
+    fi
+
+    local pips_to_install=""
+    # shellcheck disable=SC2048
+    for pkg in $*
+    do
+        local installed_version
+        if [[ "$pkg" == "git+"* ]] || ! installed_version=$(python"${pv}" "$TUE_INSTALL_SCRIPTS_DIR"/pip-correctly-installed.py "$pkg")
```

Comment:
Maybe rename the script `pip-correctly-installed` to something like `check-pip-pkg-installed-version` because right now it sounds more like its job is to check the installation of pip itself rather than the package.

Reply:
Rename done